src/Tasks/Microsoft.NET.Build.Tasks/WriteAppConfig.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class WriteAppConfig : TaskBase
+    {
+        /// <summary>
+        /// Path to the app.config source file.
+        /// </summary>
+        public ITaskItem AppConfigFile { get; set; }
+
+        [Required]
+        public string TargetFrameworkIdentifier { get; set; }
+
+        [Required]
+        public string TargetFrameworkVersion { get; set; }
+
+        /// <summary>
+        /// Path to an intermediate file where we can write the input app.config plus the generated startup supportedRuntime
+        /// </summary>
+        [Required]
+        public ITaskItem OutputAppConfigFile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            XDocument doc = LoadAppConfig(AppConfigFile);
+
+            AddSupportedRuntimeToAppconfigFile(doc, TargetFrameworkIdentifier, TargetFrameworkVersion);
+
+            if (File.Exists(OutputAppConfigFile.ItemSpec))
+            {
+                File.Delete(OutputAppConfigFile.ItemSpec);
+            }
```

Comment:
This doesn't seem necessary: FileMode.Create below will truncate the existing file.


Create | Specifies that the operating system should create a new file. If the file already exists, it will be overwritten. This requires FileIOPermissionAccess.Write permission. FileMode.Create is equivalent to requesting that if the file does not exist, use CreateNew; otherwise, use Truncate. If the file already exists but is a hidden file, an UnauthorizedAccessException exception is thrown.
-- | --

Reply:
Fixed, removed File.Exists File.Delete